src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Net.Http;
+using Microsoft.OpenApi.Any;
+using Microsoft.OpenApi.Models;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    internal static class OpenApiExtensions
+    {
+        public static string GetBasePath(this OpenApiDocument openApiDocument)
+        {
+            string basePath = null;
+            if (openApiDocument?.Servers.Count == 1)
```

Comment:
>openApiDocument?.Servers.Count == 1 [](http://example.com/codeflow?start=16&length=35)

Not super familiar with OpenApi, when would this be something other than 1?

Reply:
I have no idea, but just being defensive. @darrelmiller ?

Reply:
good link. I'll just NotImpl > 1 and then we can sort that out when we have a case.